include/pqxx/util.hxx

```diff
@@ -280,18 +280,135 @@ struct PQXX_LIBEXPORT thread_safety_model
 #  define PQXX_POTENTIAL_BINARY_ARG typename
 #endif
 
+struct byte_char_traits : std::char_traits<char>
+{
+  using char_type = std::byte;
+
+  static void assign(std::byte &a, const std::byte &b) noexcept { a = b; }
+
+  static bool eq(std::byte a, std::byte b) { return a == b; }
+
+  static bool lt(std::byte a, std::byte b) { return a < b; }
+
+  static int compare(const std::byte *a, const std::byte *b, std::size_t size)
+  {
+    for (; size; --size, ++a, ++b)
+    {
+      if (lt(*a, *b))
+        return -1;
+      if (lt(*b, *a))
+        return 1;
+    }
+
+    return 0;
+  }
+
+  static size_t length(const std::byte *data)
+  {
+    std::size_t length{};
+
+    for (; !eq(*data, std::byte{0}); ++data) ++length;
+
+    return length;
+  }
+
+  static const std::byte *
+  find(const std::byte *data, std::size_t size, const std::byte &val)
+  {
+    for (; size; --size, ++data)
+    {
+      if (eq(*data, val))
+        return data;
+    }
+
+    return nullptr;
+  }
+
+  static std::byte *
+  move(std::byte *dst, const std::byte *src, std::size_t size)
+  {
+    if (size == 0)
+      return dst;
+
+    std::byte *beginning{dst};
+
+    if (dst < src)
```

Comment:
Does `std::memmove()` make use of these traits?  If not, then why not just call that?  It may include some clever optimisation.